chemprop/cli/train.py

```diff
@@ -636,6 +639,83 @@ def build_splits(args, format_kwargs, featurization_kwargs):
     return train_data, val_data, test_data
 
 
+def summarize(args, dataset: _MolGraphDatasetMixin) -> tuple[list, list]:
+    if args.task_type in ["regression", "regression-mve", "regression-evidential"]:
+        if isinstance(dataset, MulticomponentDataset):
+            y = dataset.datasets[0].Y
+        else:
+            y = dataset.Y
+        y_mean = np.nanmean(y, axis=0)
+        print("YMEAN", y_mean)
+        y_std = np.nanstd(y, axis=0)
+        y_median = np.nanmedian(y, axis=0)
+        mean_dev_abs = np.abs(y - y_mean)
+        frac_1_sigma = np.sum((mean_dev_abs < y_std), axis=0) / len(y)
+        frac_2_sigma = np.sum((mean_dev_abs < 2 * y_std), axis=0) / len(y)
```

Comment:
Here you divide by `len(y)`. I think that will underestimate the fraction within 1 and 2 sigma when there are NaNs becaues `np.nan < number` will always be false. I think you could instead divide by the number of non NaN values. Stack exchanged offered this: `np.count_nonzero(~np.isnan(data))`.

Reply:
good catch, I changed len(y) to np.count_nonzero(~np.isnan(y))